app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt

```diff
@@ -0,0 +1,253 @@
+            "Nix",			// An interesting spelling of a word meaning "nothing" in Dutch
+            "Nyle",			// An interesting spelling of the great river
+            "Odet",
+            "Og",			// The last of the witches in Pratchett's Discworld
+            "Omega",		// The end of it all
+            "Omphalos",		// Greek for navel
+            "Orias",
+            "Othello",		// From Shakespeare
+            "Parade",		// This word means the same in Dutch and in English
+            "Penthara",
+            "Picard",		// The enigmatic captain from ST:TNG
+            "Pollux",		// Brother of Castor
+            "Quator",
+            "Rakhar",
+            "Ran",			// A film by Akira Kurosawa
+            "Regulas",
+            "Relva",
+            "Rhymus",
+            "Rochani",
+            "Rubicum",		// The river Ceasar crossed to get into Rome
+            "Rutia",
+            "Sarpeidon",
+            "Sefalla",
+            "Seltrice",
+            "Sigma",
+            "Sol",			// That's our own solar system
+            "Somari",
+            "Stakoron",
+            "Styris",
+            "Talani",
+            "Tamus",
+            "Tantalos",		// A king from a Greek tragedy
+            "Tanuga",
+            "Tarchannen",
+            "Terosa",
+            "Thera",		// A seldom encountered Dutch girl's name
+            "Titan",		// The largest moon of Jupiter
+            "Torin",		// A hero from Master of Magic
+            "Triacus",
+            "Turkana",
+            "Tyrus",
+            "Umberlee",		// A god from AD&D, which has a prominent role in Baldur's Gate
+            "Utopia",		// The ultimate goal
+            "Vadera",
+            "Vagra",
+            "Vandor",
+            "Ventax",
+            "Xenon",
+            "Xerxes",		// A Greek hero
+            "Yew",			// A city which is in almost all of the Ultima games
+            "Yojimbo",		// A film by Akira Kurosawa
+            "Zalkon",
+            "Zuul",		// From the first Ghostbusters movie
+            "Gorgon",
+            "Hydra",
+            "Phoenix",
+            "Gagrin",
+            "Athens",
+            "Knossos",
+            "Macedon",
+            "Thorne",
+            "Sparta",
+            "Dis",
+            "Farinata",
+            "Century",
+            "Theseus",
+            "Asgard"
+        )
+    }
+
+    /**
+     * Generates universe with constant number of solar systems. Each solar system can have one to three planets.
+     *
+     */
+    fun generateUniverse() {
+        var i: Int = NUM_SYSTEMS
+        while (i > 0) {
+            val x = random.nextInt(X_SIZE)
+            val y = random.nextInt(Y_SIZE)
+            val num_planets = random.nextInt(MAX_PLANETS) + 1
+            val pair = Pair(x, y)
+            if (!universe.containsKey(pair)) {
+                val planetSet = generatePlanets(num_planets)
+                val systemName: String = namesList[random.nextInt(namesList.size)]
```

Comment:
could you still get a solar system with the same names or a planet and solar system with the same name? I don't see where the solar system's name is also removed afterwards

Reply:
yes, fixing it now